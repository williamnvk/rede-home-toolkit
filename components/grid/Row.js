import React from 'react';
import { Component } from './styled';
import { defaultDirection } from '../../constants/flex';

function renderRow(props) {
  const p = {
    ...props,
    direction: props.direction || defaultDirection,
  };
  return <Component {...p} />;
}

export default React.memo(renderRow);
