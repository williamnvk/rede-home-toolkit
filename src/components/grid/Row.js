// @flow
import React, { memo } from 'react';
import { Grid } from './styled';
import type { RowProps } from './types';
import { withRef } from '../../utils';

function renderRow(props: RowProps) {
  return <Grid el="row" {...props} />;
}

export default memo<RowProps>(withRef(renderRow));
