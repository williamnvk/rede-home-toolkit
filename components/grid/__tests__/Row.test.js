import React from "react"
import { shallow } from 'enzyme';
import Row from '../Row.js';

describe('Row', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Row />);
    expect(wrapper).toMatchSnapshot();
  });
});
