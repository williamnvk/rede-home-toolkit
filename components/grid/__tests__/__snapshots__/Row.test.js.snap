// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders correctly 1`] = `ShallowWrapper {}`;
